remix/app/routes/_app.newsletter.create.tsx

```diff
@@ -0,0 +1,59 @@
+import { ActionArgs, json } from '@remix-run/node';
+import { Form } from '@remix-run/react';
+import { withZod } from '@remix-validated-form/with-zod';
+import { validationError } from 'remix-validated-form';
+import { z } from 'zod';
+
+import appHandler from '../lib/handler/app.handler';
+import NewsletterRepository from '../repositories/newsletter.server';
+
+const validator = withZod(
+  z.object({
+    name: z.string().min(1, { message: 'Name is required' }),
+    content: z.string().min(1, { message: 'Content is required' }),
+  })
+);
+
+export async function action({ request }: ActionArgs) {
+  return appHandler(request, async (user) => {
+    const result = await validator.validate(await request.formData());
+
+    if (result.error) {
+      return validationError(result.error);
+    }
+
+    const newsletterData = {
+      ...result.data,
+      userId: user.id,
+    };
+
+    try {
+      const newsletter = await NewsletterRepository.create({
+        data: newsletterData,
+      });
+
+      return json({ ...newsletter });
+    } catch (error) {
+      throw error;
+    }
```

Comment:
If we remove this `try` `catch` block, I think the code will still behave the same as before, right? 👀

Reply:
Um, Seem like this become my behavior, I was too worry about the Error.

Next time I think more before adding `try``catch`.

Removed here: [4c0b609](https://github.com/nimblehq/js-framework-benchmark/pull/112/commits/4c0b60954c211a006e6d02bac67e815fe2de8826)